src/config.c

```diff
@@ -801,68 +840,76 @@ void configSetCommand(client *c) {
         }
 
         /* If this config appears twice then fail */
-        for (j = 0; j < i; j++) {
-            if (set_configs[j] == config) {
-                /* Note: we don't abort the loop since we still want to handle redacting sensitive configs (above) */
-                errstr = "duplicate parameter";
-                err_arg_name = c->argv[2+i*2]->ptr;
-                invalid_args = 1;
-                break;
-            }
+        sds config_name = sdsnew(config->name);
+        if (dictFind(set_configs, config_name)) {
+            /* Note: we don't abort the loop since we still want to handle redacting sensitive configs (above) */
+            errstr = "duplicate parameter";
+            err_arg_name = c->argv[2+i*2]->ptr;
+            invalid_args = 1;
+        } else {
+            dictAdd(set_configs, sdsnew(config->name), config);
+            dictAdd(new_values, sdsnew(config->name), c->argv[2+i*2+1]->ptr);
```

Comment:
It seems to me that for every entry in `set_config` we have an entry in `new_values` and an entry in `old_values`. It might be best to construct a dict we a value containing a 3 entry struct:
```c
typedef struct {
   standardConfig* config;
   sds old_value;
   sds new_value;
} configSetValue;
```

Reply:
Agreed, this seems like a good idea. I'll change the code to follow this